File: ui/src/app/dto/campaign-condition-dto.ts
export class CampaignConditionDto {
  constructor(type: number) {
    this.type = type;
  }

  // See AppConstants.CAMPAIGN
  id?: number;
  type: number;
  target!: number;
  stage?: number;
  scheduleDate?: Date;
  scheduleHour?: number;
  scheduleMinute?: number;
  operation?: number;
  value?: string;
  propertyName?: string;

}

File: esthesis-core-backend/services/srv-campaign/srv-campaign-client/src/main/java/esthesis/service/campaign/dto/CampaignConditionDTO.java
package esthesis.service.campaign.dto;

import esthesis.common.AppConstants;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class CampaignConditionDTO {

  private AppConstants.Campaign.Condition.Type type;
  private AppConstants.Campaign.Condition.Op operation;
  private AppConstants.Campaign.Condition.Stage stage;
  private int group;

  private String value;
  private String propertyName;
  private Boolean propertyIgnorable;
  private Instant scheduleDate;
}
